vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java

```diff
@@ -105,6 +140,31 @@ public <TValueProperty> ComboBoxBase(String valuePropertyName,
         dataController = new ComboBoxDataController<>(this, this::getLocale);
         dataController.getDataGenerator().addDataGenerator((item,
                 jsonObject) -> jsonObject.put("label", generateLabel(item)));
+
+        // Configure web component to use key property from the generated
+        // wrapper items for identification
+        getElement().setProperty("itemValuePath", "key");
+        getElement().setProperty("itemIdPath", "key");
+
+        // Disable template warnings
+        getElement().setAttribute("suppress-template-warning", true);
+
+        // Synchronize input element value property state when setting a custom
+        // value. This is necessary to allow clearing the input value in
+        // `ComboBox.refreshValue`. If the input element value is not
+        // synchronized here, then setting the property to an empty value would
+        // not trigger a client update. Need to use
+        // `addInternalCustomValueSetListener` here, in order to avoid enabling
+        // custom values, which is a side effect of
+        // `ComboBoxBase.addCustomValueSetListener`.
+        addInternalCustomValueSetListener(e -> getElement()
```

Comment:
Added a new internal method `addInternalCustomValueSetListener` to avoid tracking the custom value listener.